### DISCUSSION TITLE
How to controll my Modbus Roaster through Artisan

### DISCUSSION THREAD

**Discussion by adwaniah** [artisan-roaster-scope/artisan#1247] [Q&A]
Hello,

I have successfully connected my roaster to Artisan and the readings are OK also the curves.
I need to control my roaster using Artisan (burner, Fan, etc.).

Can anyone help?, please

Thanks.

**Comment by roasterdave** [comment]
Every roaster is different.  Easiest is to use Artisan's machine setup (Config>> Machine) to do the configuration.  The [Machines](https://artisan-scope.org/machines/) page on [artisan-scope.org](https://artisan-scope.org) has specific information and details for those roasters that can be controlled.  

If your roaster is not in this list you will need to research its Modbus registers and addresses.  Then write the appropriate Modbus commands to control it from the sliders or buttons.  Click the Help button on Config>> Events>> Sliders tab or Buttons tab to see details about the available Modbus commands.  Examples can be found on the [Artisan blog](https://artisan-roasterscope.blogspot.com/).  Start with a search for 'Modbus'.  If you plan to use PID control, first get manual control working before you try to tackle the PID.

**Reply by adwaniah** [reply]
Thank very much for your help,
my machine is listed on Artisan, but unfortunately there is no specific documentation.

**Reply by MAKOMO** [reply]
Well it could help us if you name us your machine and best even the Artisan setup you are running.

**Comment by adwaniah** [comment]
![Full_screen](https://github.com/artisan-roaster-scope/artisan/assets/141586335/4de4a283-1de1-40ac-9cb6-73464087e165)
![Full_screen_w_controlls](https://github.com/artisan-roaster-scope/artisan/assets/141586335/41c8b4dd-eba0-4f5f-9bea-4168a9b0d188)
![machines_list](https://github.com/artisan-roaster-scope/artisan/assets/141586335/c4199c52-5e82-4f09-a67a-27681c349022)
![port_settings](https://github.com/artisan-roaster-scope/artisan/assets/141586335/7af230af-f09e-430a-8ba4-ae586017cf21)
![Version](https://github.com/artisan-roaster-scope/artisan/assets/141586335/84bccb33-d83a-4bfb-9b2f-01f9c768c618)

**Comment by adwaniah** [comment]
my machine is https://goldenroaster.com/gr-3-mini/ full automatic,
as shown on my attachments, port settings (working fine),

is there a way to control my machine through Artisan (Burner/Air/etc.)?
also can I connect my machine using Ethernet using UDP/TCP?

**Reply by bigl00z3** [reply]
Have you checked [https://artisan-scope.org/machines/goldenroasters/](https://artisan-scope.org/machines/goldenroasters/) it mentions you need a driver?

On your roaster's page  [https://goldenroaster.com/gr-3-mini/ ](https://goldenroaster.com/gr-3-mini/ )I see: 

> - Remote connection and control with Wireless Communication.
> - Wireless can be monitored and controlled remotely from any phone, tablet or computer with Android, IPhone, Ipad and Windows.
> - Datalogger feature, temperature and time graph can be followed while coffee is roasted. Past data records can be seen.
> - Automatic or manual control possibility. Automatic operation according to temperature graph.

So it should be possible.

When you connect with your current settings, does it read the values (ET, BT...)correctly? 
What about if you click on one of the control buttons (SV) does it change the target temperature on the roaster?

If those already work, you could add more buttons or sliders in `Config/Events` and point them to control the right Modbus port (you have a HELP button at the bottom of the Events window to give you commands explanations and examples).

Nico

**Reply by adwaniah** [reply]
Thank you for the reply,
Yes, my machine is working fine full automatic, 
but I need to connect it to Artisan for better graph & view (laptop monitor/PC), etc.
also it is working on Artisan as logging only,
but as you mentioned Yes, if I click on one of the control buttons (SV) it changes the target temperature on the roaster.

I need to see some examples for modbus commands so I can add accordingly, Please
also I have one issue with Artisan when I click "OFF" button the  the USB disconnects and will not work until I remove it from the port and connect it again, Please help.

Thanks.

**Comment by MAKOMO** [comment]
The blue box on https://artisan-scope.org/machines/index lists machines that Artisan is able to control. Your machine is not listed. It might be technical possible that your machine can be controlled, but the Artisan team has no information on how this could be achieved (and potentially it is not achievable with Artisan), mostly as the producer of your machine is not supporting the Artisan project. Ask GoldenRoast for support or ask them to support the Artisan project!

**Comment by adwaniah** [comment]
Thank you for the reply,
Yes, my machine is working fine full automatic, 
but I need to connect it to Artisan for better graph & view (laptop monitor/PC), etc.
also it is working on Artisan as logging only,
when I click on one of the control buttons (SV) it changes the target temperature on the roaster.

I need to see some examples for modbus commands so I can add accordingly, Please
also I have one issue with Artisan when I click "OFF" button the  the USB disconnects and will not work until I remove it from the port and connect it again, Please help.

Thanks.

**Reply by bigl00z3** [reply]
You can see a couple MODBUS commands example in my blog post [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf)  and then the description for all the possibilities is available in the `Help` window located in the `Config/Events` page.
![image](https://github.com/artisan-roaster-scope/artisan/assets/8886922/69224b7a-f37b-42bc-95ed-1a4bf6b31926)
![image](https://github.com/artisan-roaster-scope/artisan/assets/8886922/f385e21b-d4c8-42e0-ac07-efabfa6269fb)


You can send command to Modbus via **Buttons** or **Sliders** (both in `Config/Events`) or also if you create **Alarms** (in `Config/Alarms`) that are calling a specific **Button** number
Nico

**Reply by poundy** [reply]
I'd like to make doubly sure that you are aware of the support available to you. The artisan project is operated on a donation basis, and done by volunteers. Manufacturers assist adding their machines in those supported in Artisan by funding the integration effort necessary.  Your manufacturer has not engaged with the project.  

As Marko said earlier: The blue box on https://artisan-scope.org/machines/index lists machines that Artisan is able to control. Your machine is not listed. It might be technical possible that your machine can be controlled, but the Artisan team has no information on how this could be achieved (and potentially it is not achievable with Artisan), mostly as the producer of your machine is not supporting the Artisan project. Ask GoldenRoast for support or ask them to support the Artisan project!

Nobody here has access to your roaster, nobody here will be able to assist very deeply. You should approach your hardware supplier and ask them to get your machine supported fully.
